realm-helpers/src/main/java/chat/rocket/android/realm_helper/RealmObjectObserver.java

```diff
@@ -47,29 +47,31 @@ public void setImpl(Impl<T> impl) {
 
   private T previousResult;
 
-  @Override protected final RealmResults<T> queryItems(Realm realm) {
+  @Override
+  protected final RealmResults<T> queryItems(Realm realm) {
     return query.query(realm).findAll();
   }
 
-  @Override protected final RealmChangeListener<RealmResults<T>> getListener() {
+  @Override
+  protected final RealmChangeListener<RealmResults<T>> getListener() {
     return new RealmChangeListener<RealmResults<T>>() {
-      @Override public void onChange(RealmResults<T> element) {
-        T source = impl.extractObjectFromResults(element);
-        T target = source != null ? realm.copyFromRealm(source) : null;
-        if (previousResult != null && impl.isSame(previousResult, target)) {
+      @Override
+      public void onChange(RealmResults<T> element) {
+        T currentResult = impl.extractObjectFromResults(element);
+        if (previousResult != null && impl.isSame(previousResult, currentResult)) {
           return;
         }
-        previousResult = target;
+        previousResult = currentResult;
```

Comment:
I suspect it is dangerous to preserve the Realm object without detaching from DB...